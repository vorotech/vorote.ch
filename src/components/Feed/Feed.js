// @flow strict
import React from 'react';
import { Link } from 'gatsby';
import moment from 'moment';
import styled from 'styled-components';
import type { Edges } from '../../types';
import { marginTop, marginBottom } from '../../utils/get-margin';
import { lineHeight } from '../../utils/get-line-height';

const FeedItem = styled.div`
  ${marginTop(1.25)}
  &:last-child {
    ${marginBottom(0.5)}
  }
`;

const MetaTime = styled.time`
  font-size: ${(props) => props.theme.fontSizes.small}px;
  color: var(--color-text);
  font-weight: ${(props) => props.theme.fontWeights.heading};
  text-transform: uppercase;
`;

const MetaDevider = styled.span`
  margin: 0 5px;
`;

const MetaCategoryLink = styled(Link)`
  font-size: ${(props) => props.theme.fontSizes.small}px;
  color: var(--color-secondary);
  font-weight: ${(props) => props.theme.fontWeights.heading};
  text-transform: uppercase;

  &:hover,
  &:focus {
    color: var(--color-primary);
  }
`;

const Title = styled.h2`
  font-size: ${(props) => props.theme.fontSizes.base * 1.6875}px;
  ${lineHeight(1.5)}
  margin-top: 0;
  ${marginBottom(0.5)}
`;

const TitleLink = styled(Link)`
  color: var(--color-text);

  &:hover,
  &:focus {
    color: var(--color-text);
    border-bottom: 1px solid var(--color-text);
  }
`;

const Description = styled.p`
  font-size: ${(props) => props.theme.fontSizes.base}px;
  ${lineHeight(1)}
  ${marginBottom(0.75)}
`;

const ReadMoreLink = styled(Link)`
  font-size: ${(props) => props.theme.fontSizes.base}px;
  color: var(--color-primary);

  &:hover,
  &:focus {
    color: var(--color-primary);
    border-bottom: 1px solid var(--color-primary);
  }
`;

type Props = {
  edges: Edges
};

const Feed = ({ edges }: Props) => (
  <div>
    {edges.map((edge) => (
      <FeedItem key={edge.node.fields.slug}>
        <div>
          <MetaTime dateTime={moment(edge.node.frontmatter.date).format('MMMM D, YYYY')}>
            {moment(edge.node.frontmatter.date).format('MMMM YYYY')}
          </MetaTime>
          <MetaDevider />
          <span>
            <MetaCategoryLink to={edge.node.fields.categorySlug}>
              {edge.node.frontmatter.category}
            </MetaCategoryLink>
          </span>
        </div>
        <Title>
          <TitleLink to={edge.node.fields.slug}>{edge.node.frontmatter.title}</TitleLink>
        </Title>
        <Description>{edge.node.frontmatter.description}</Description>
        <ReadMoreLink to={edge.node.fields.slug}>Read post</ReadMoreLink> <span>ðŸ•’ {edge.node.fields.readingTime.text}</span>
      </FeedItem>
    ))}
  </div>
);

export default Feed;
