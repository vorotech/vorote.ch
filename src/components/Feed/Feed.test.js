// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import Feed from './Feed';
import 'jest-styled-components';
import Theme from '../Theme';

describe('Feed', () => {
  const props = {
    edges: [
      {
        node: {
          fields: {
            slug: '/test_0',
            categorySlug: '/test_0',
            tagSlugs: [
              '/test-1',
              '/test-2'
            ],
            readingTime: {
              text: '1 min read'
            }
          },
          frontmatter: {
            date: '2016-09-01',
            description: 'test_0',
            category: 'test_0',
            tags: [
              'test-1',
              'test-2'
            ],
            title: 'test_0'
          },
          id: 'test-123',
          html: '<p>test</p>'

        }
      },
      {
        node: {
          fields: {
            slug: '/test_1',
            categorySlug: '/test_1',
            tagSlugs: [
              '/test-1',
              '/test-2'
            ],
            readingTime: {
              text: '1 min read'
            }
          },
          frontmatter: {
            date: '2016-09-01',
            description: 'test_1',
            category: 'test_1',
            tags: [
              'test-1',
              'test-2'
            ],
            title: 'test_1'
          },
          id: 'test-321',
          html: '<p>test</p>'

        }
      }
    ]
  };

  it('renders correctly', () => {
    const tree = renderer.create(
      <Theme>
        <Feed {...props} />
      </Theme>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
