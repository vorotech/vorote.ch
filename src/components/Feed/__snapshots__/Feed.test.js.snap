// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed renders correctly 1`] = `
.c0 {
  margin-top: 32.5px;
}

.c0:last-child {
  margin-bottom: 13px;
}

.c1 {
  font-size: 14px;
  color: var(--color-text);
  font-weight: 600;
  text-transform: uppercase;
}

.c2 {
  margin: 0 5px;
}

.c3 {
  font-size: 14px;
  color: var(--color-secondary);
  font-weight: 600;
  text-transform: uppercase;
}

.c3:hover,
.c3:focus {
  color: var(--color-primary);
}

.c4 {
  font-size: 27px;
  line-height: 2.4375em;
  margin-top: 0;
  margin-bottom: 13px;
}

.c5 {
  color: var(--color-text);
}

.c5:hover,
.c5:focus {
  color: var(--color-text);
  border-bottom: 1px solid var(--color-text);
}

.c6 {
  font-size: 16px;
  line-height: 1.625em;
  margin-bottom: 19.5px;
}

.c7 {
  font-size: 16px;
  color: var(--color-primary);
}

.c7:hover,
.c7:focus {
  color: var(--color-primary);
  border-bottom: 1px solid var(--color-primary);
}

<div>
  <div
    className="c0"
  >
    <div>
      <time
        className="c1"
        dateTime="September 1, 2016"
      >
        September 2016
      </time>
      <span
        className="c2"
      />
      <span>
        <a
          className="c3"
          href="/test_0"
        >
          test_0
        </a>
      </span>
    </div>
    <h2
      className="c4"
    >
      <a
        className="c5"
        href="/test_0"
      >
        test_0
      </a>
    </h2>
    <p
      className="c6"
    >
      test_0
    </p>
    <a
      className="c7"
      href="/test_0"
    >
      Read post
    </a>
    <span>
      ðŸ•’ 
      1 min read
    </span>
  </div>
  <div
    className="c0"
  >
    <div>
      <time
        className="c1"
        dateTime="September 1, 2016"
      >
        September 2016
      </time>
      <span
        className="c2"
      />
      <span>
        <a
          className="c3"
          href="/test_1"
        >
          test_1
        </a>
      </span>
    </div>
    <h2
      className="c4"
    >
      <a
        className="c5"
        href="/test_1"
      >
        test_1
      </a>
    </h2>
    <p
      className="c6"
    >
      test_1
    </p>
    <a
      className="c7"
      href="/test_1"
    >
      Read post
    </a>
    <span>
      ðŸ•’ 
      1 min read
    </span>
  </div>
</div>
`;
